modules/distribution/resources/migration-1.8.0_to_1.9.0/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom18to19.java

```diff
@@ -0,0 +1,702 @@
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.*;
+import java.util.*;
+
+
+/**
+ * This class contains all the methods which is used to migrate APIs from APIManager 1.8.0 to APIManager 1.9.0.
+ * The migration performs in database, registry and file system
+ */
+
+@SuppressWarnings("unchecked")
+public class MigrateFrom18to19 implements MigrationClient {
+
+    private static final Log log = LogFactory.getLog(MigrateFrom18to19.class);
+    private List<Tenant> tenantsArray;
+
+    public MigrateFrom18to19() throws UserStoreException {
+        TenantManager tenantManager = ServiceHolder.getRealmService().getTenantManager();
+        tenantsArray = new ArrayList(Arrays.asList(tenantManager.getAllTenants()));
+        Tenant superTenant = new Tenant();
+        superTenant.setDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        superTenant.setId(MultitenantConstants.SUPER_TENANT_ID);
+        tenantsArray.add(superTenant);
+    }
+
+    /**
+     * This method is used to migrate database tables
+     * This executes the database queries according to the user's db type and alters the tables
+     *
+     * @param migrateVersion version to be migrated
+     * @throws APIMigrationException
+     * @throws SQLException
+     */
+    @Override
+    public void databaseMigration(String migrateVersion) throws APIMigrationException, SQLException {
+        log.info("Database migration for API Manager 1.8.0 started");
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            String queryToExecute = ResourceUtil.pickQueryFromResources(migrateVersion, Constants.ALTER).trim();
+            connection = APIMgtDBUtil.getConnection();
+
+            String queryArray[] = queryToExecute.split(Constants.LINE_BREAK);
+
+            connection.setAutoCommit(false);
+
+            for (String query : queryArray) {
+                preparedStatement = connection.prepareStatement(query);
+                preparedStatement.execute();
+                connection.commit();
+                preparedStatement.close();
+            }
+
+            String dbType = MigrationDBCreator.getDatabaseType(connection);
+            dropFKConstraint(migrateVersion, dbType);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Query " + queryToExecute + " executed ");
+            }
+
+        } catch (IOException e) {
+            //ResourceUtil.handleException("Error occurred while finding the query. Please check the file path.", e);
+            log.error("Error occurred while migrating databases", e);
+        } catch (Exception e) {
+            //ResourceUtil.handleException("Error occurred while finding the query. Please check the file path.", e);
+            log.error("Error occurred while migrating databases", e);
```

Comment:
If any error occurs while migrating the databases, the process will log the error without terminating. The reason is to do so is, suppose one tried with already migrated database then the user will not be able to migrate other resources such as registry/file system.
No exceptions will be thrown, and I will remove the thrown ones.